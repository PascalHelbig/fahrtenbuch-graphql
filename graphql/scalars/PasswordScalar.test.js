const { makeExecutableSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const PasswordScalar = require('./PasswordScalar');

const testPassword = (password) => {
  const SchemaDefinition = `
    type Query { password(input: Password!) : Password }
    schema { query: Query }
  `;
  const resolvers = {
    Query: {
      password: (_, { input }) => input,
    },
    Password: PasswordScalar.resolver,
  };
  const schema = makeExecutableSchema({
    typeDefs: [SchemaDefinition, PasswordScalar.schema],
    resolvers,
  });

  const query = `query { password (input: "${password}") }`;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot()
  );
};

it('should except a password with 5 chars', () =>
  testPassword('12345')
);

it('should not except a password with less then 5 chars', () =>
  testPassword('1234')
);
