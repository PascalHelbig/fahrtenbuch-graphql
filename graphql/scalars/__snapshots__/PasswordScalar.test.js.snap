exports[`test should except a password with 5 chars 1`] = `
Object {
  "data": Object {
    "password": "12345",
  },
}
`;

exports[`test should not except a password with less then 5 chars 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Query error: minimum length of 5 required: ],
  ],
}
`;

exports[`test should not except a password with more than 254 chars 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Query error: maximum length is 254: ],
  ],
}
`;
